text/0000-mixed-shadow-mode.md

```diff
@@ -0,0 +1,175 @@
+
+### Instrumentation
+
+Applications that rely on instrumentation libraries that don't yet support Shadow DOM are currently
+able to obtain references to descriptors like `addEventListener()` before they are patched and use
+them to override `@lwc/synthetic-shadow` polyfills where needed. Such workarounds would not work for
+events originating from components that prefer native Shadow DOM.
+
+### Styling
+
+Applications currently rely on global stylesheets to apply deeply throughout the DOM. With native
+Shadow DOM, this would not be possible. All of a component's styles would have to be generated and
+added to the component bundle.
+
+### Testing
+
+LWC currently has integration tests for synthetic shadow mode and native shadow mode. A third
+testing mode called "mixed shadow mode" will be introduced. Synthetic shadow DOM polyfills are
+implemented as patches on global prototypes so the decision of invoking a polyfilled API vs a native
+API will depend on the preference of individual components during runtime. This test environment
+will ensure that components designed to run in a native Shadow DOM environment will work correctly
+when even when synthetic Shadow DOM polyfills are present.
+
+Components that prefer native mode will need to also support synthetic mode in browser environments
+where Shadow DOM is natively unavailable, so they will also need to be tested in both modes. LWC
+test utilities will need to be updated to facilitate this.
+
+The existing WPT (Web Platform Tests) test suite for Shadow DOM APIs will also be run to identify
+any coverage gaps in LWC integration tests.
+
+## Observable differences
+
+### Light DOM and assigned elements
+
+In native Shadow DOM, elements are rendered in a component's light DOM and assigned to slots in a
+child component's shadow DOM. This means that elements will exist in the DOM regardless of whether
+they are assigned to a slot. This is not the case for LWC--elements that are passed down to child
+components are never rendered unless they are assigned to a slot.
+
+In the following example, `span` will not exist in the DOM in synthetic shadow, but it will exist in
+native shadow.
+
+```html
+<!-- x-parent -->
+<template>
+    <x-child>
+        <span>foo</span>
+    </x-child>
+</template>
+
+<!-- x-child -->
+<template>
+    <p>child</p>
+</template>
+```
+
+This observable difference will not be remediated as such a change would be non-trivial and
+backwards-incompatible with the current synthetic shadow implementation.
+
+### this.shadowRoot vs this.template
+
+`this.shadowRoot` references the component's shadow root in native Shadow DOM while `this.template`
+references the component's shadow root in synthetic Shadow DOM. This observable difference can be
+remediated for components that favor native Shadow DOM by aliasing `this.template` to
+`this.shadowRoot`. There are no plans to enable `this.shadowRoot` for existing components as the
+current LWC-ism is to use `this.template`.

```

Comment:
This section is a bit unclear to me. Are we planning to make `this.template` evaluate to `null` in native shadow mode?

Reply:
>  while `this.template` references the component's shadow root in synthetic Shadow DOM.

This isn't entirely true, when `@lwc/synthetic-shadow` isn't loaded `this.template` refers to the native shadow root.

Reply:
Fixed, thanks.